Tensile/Source/client/main.cpp

```diff
@@ -470,6 +492,41 @@ namespace Tensile
     } // namespace Client
 } // namespace Tensile
 
+size_t calculate_flush_count(size_t arg_flush_count,
+                                   size_t arg_flush_memory_size,
+                                   Tensile::Client::ClientProblemFactory const& problemFactory)
+{
+    size_t default_arg_flush_count = 1;
+    size_t default_arg_flush_memory_size = 0;
+    size_t flush_count             = default_arg_flush_count;
+
+    if(arg_flush_count != default_arg_flush_count
+       && arg_flush_memory_size != default_arg_flush_memory_size)
+    {
+        std::cout << "Tensile WARNING: cannot set both flush_count and flush_memory_size"
+                     << std::endl;
+        std::cout << "Tensile WARNING: using flush_count = " << arg_flush_count
+                     << std::endl;
+        flush_count = arg_flush_count;
+    }
+    else if(arg_flush_count != default_arg_flush_count)
+    {
+        flush_count = arg_flush_count;
+    }
+    else if(arg_flush_memory_size != default_arg_flush_memory_size)
+    {
+        size_t cached_size = 0;
+
+        for(auto const& problem : problemFactory.problems())
+            cached_size = std::max(cached_size, problem.a().sizes()[0]*problem.a().sizes()[1]*problem.a().elementBytes() +
+                                                problem.b().sizes()[0]*problem.b().sizes()[1]*problem.b().elementBytes() +
+                                                problem.c().sizes()[0]*problem.c().sizes()[1]*problem.c().elementBytes());
+
+        flush_count = 1 + (arg_flush_memory_size - 1) / cached_size;
```

Comment:
rocBLAS adds some prints here to describe flush count used. Do we want to add that here too?

https://github.com/ROCm/rocBLAS-internal/blob/51a1056edd56a6f0a3f1e40f2f2eb0b2e4138f4f/clients/common/utility.cpp#L469

Reply:
I think it is better to print it in case we specify either flush_count or flush_memory_size.